manifests/01-deployment.yaml

```diff
@@ -0,0 +1,86 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: etcd-quorum-guard
+  namespace: kube-system
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      name: etcd-quorum-guard
+  strategy:
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: etcd-quorum-guard
+        app: etcd-quorum-guard
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - "etcd-quorum-guard"
+            topologyKey: kubernetes.io/hostname
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      priorityClassName: "system-cluster-critical"
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+        operator: Exists
+      - key: node.kubernetes.io/memory-pressure
+        effect: NoSchedule
+        operator: Exists
+      - key: node.kubernetes.io/disk-pressure
+        effect: NoSchedule
+        operator: Exists
+      - key: node.kubernetes.io/not-ready
+        effect: NoExecute
+        operator: Exists
+      - key: node.kubernetes.io/unreachable
+        effect: NoExecute
+        operator: Exists
+      - key: node.kubernetes.io/unschedulable
+        effect: NoExecute
+        operator: Exists
+      containers:
+      - image: registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+        imagePullPolicy: IfNotPresent
+        name: etcd-quorum-guard-container
+        volumeMounts:
+        - mountPath: /mnt/kube
+          name: kubecerts
+        command: "/bin/sleep"
+        args:
+        - "infinity"
+        readinessProbe:
+          exec:
+            command:
+            - /bin/sh
+            - -c
+            - |
+                declare -r croot=/mnt/kube
+                declare -r health_endpoint="https://10.128.0.1:2379/health"
```

Comment:
Is the hard-coded IP address because you don't want to rely upon DNS? Is the address the same across all clusters a customer can install?

Reply:
That's one of my big concerns here.  That is default address, but it can (at least in principle) be changed.  I'm fine with relying on DNS (if it works), but I haven't found an address that works here.

If I debug the node, I can just use localhost, but the pod of course has virtualized networking.

Reply:
I don't know how reliable this is, but I was able to resolve the hostname contained in the certificate filename from inside a pod. You could extract the hostname from certificate filename like this:
```
host=${cert##*system:etcd-peer:}
host=${host%.crt}
```
So while it removes the hard-coded IP address and lets you get rid of the `curl -k` and `--resolve`, it does rely upon DNS working to pass the health check.